### ISSUE TITLE
Functions:  FunctionParameterPropertyValue

### ISSUE THREAD

**Issue by paleicikas** [betalgo/openai#304]
Is there a bug related to FunctionParameterPropertyValue types. It appears that only the "string" type is supported, as it doesn't seem to work for "array".

`The JSON value could not be converted to System.String. Path: $.error.message | LineNumber: 2 | BytePositionInLine: 16.`

**Comment by eqqe** [comment]
I opened a PR for this issue https://github.com/betalgo/openai/pull/307

**Comment by rzubek** [comment]
Since C# doesn't support discriminated union types (yet), there isn't a great solution. 

We can change the type of FunctionParameterPropertyValue.Type to `object` and hope client code does the right thing?

Slightly orthogonally, but not entirely: I'm not sure that specifying union types to ChatGPT will actually be understood and processed. Tellingly, none of OpenAI's examples use discriminated types.

**Comment by rzubek** [comment]
On a second reading, I think I completely misunderstood your post. I thought you were asking about union types - but on further re-reading, it looks like you were actually asking about having an argument be "array" instead of "number" or "scalar" or other such, right?

The fix for this is in PR #320.

There were two issues:

1. The error "The JSON value could not be converted to System.String" was actually a bug with deserializing error messages from the server, which was unrelated to arrays :)

2. Once that was resolved, I added support for arrays using a new function AddArrayParameter(), used like this:
```
       var fn4 = new FunctionDefinitionBuilder("identify_number_sequence", "Get a sequence of numbers present in the user message")
            .AddArrayParameter("values", "number", "Sequence of numbers specified by the user")
            .Build();
```
The issue here is that it looks like arrays require additional information about the type of their items, so there is a new function that lets client code specify item type ("number" in this case).
